src/test/resources/com/atomist/rug/test/gherkin/handler/event/PassingFeature1StepsAgainstGeneratedWithArrays.ts

```diff
@@ -0,0 +1,17 @@
+import {Given,When,Then, HandlerScenarioWorld} from "@atomist/rug/test/handler/Core"
+
+import {Push} from "@atomist/rug/cortex/stub/Push"
+import {Commit} from "@atomist/rug/cortex/stub/Commit"
+
+ 
+Given("a sleepy country", f => {
+})
+When("a visionary leader enters", world => {
+   world.registerHandler("ReturnsEmptyPlanEventHandlerGenWithArrays")
+   let p = new Push().addContains(new Commit())
```

Comment:
Okay, let's learn for a while before we can see how to have something more intuitive and, should I say developer friendly.